### PR TITLE
Ensure that `src/scripts/ncbi_postgres_nmdc_exact_term_matching.py` reads credentials from `local/.env`

### PR DESCRIPTION
**Pull request by sujaypatil96**
This PR is a continuation of the work in #1931 

Things that are being addressed in this PR:
* renaming `ncbi_nmdc_exact_term_matching.py` --> `ncbi_postgres_nmdc_exact_term_matching.py`
  * reason: there will be another TSV mapping file coming in from #1940 and the script that produces that mapping file should be called ncbi_nmdc_exact_term_matching.py because that maps fields/slots from the NCBI Attributes XML (not the NCBI postgres database) to NMDC slots
* make sure `src/scripts/ncbi_postgres_nmdc_exact_term_matching.py` follows best practices and reads NCBI postgres database connection credentials from `local/.env` file

### CODE REVIEW

File: assets/misc/.env.template [hunk 1]
```diff
@@ -13,3 +13,6 @@ DEST_MONGO_PASS=
 DEST_MONGO_HOST=
 DEST_MONGO_PORT=
 DEST_MONGO_AUTH_SUFFIX=authSource=admin&readPreference=primary&directConnection=true&ssl=false
+
+NCBI_POSTGRES_USER=
```
Comment (turbomam) [inline assets/misc/.env.template:17]: I'm not crazy about these constant names. They imply that they provide access to a Postgres server maintained by NCBI. Really, it's  Postgres database containing NCBI Biosample data, maintained by the BBOP group/NMDC

I'll try to think of something more helpful

If you want to change them now, how about just POSTGRES_* ?

**Review by turbomam — CHANGES_REQUESTED** [review]

File: assets/misc/.env.template (hunk 1 above)
Comment (sujaypatil96) [reply assets/misc/.env.template:17]: Everything we are doing is in the context of NMDC, with that context in mind, I think it's possible that just POSTGRES_* might lead some people (users, developers) to be confused between the Postgres database that `nmdc-server` (data portal) read from, and the Postgres database containing NCBI BioSample data?

How about something like NCBI_DB_AS_POSTGRES_*?

Comment (turbomam) [reply assets/misc/.env.template:17]: and no mention of biosample?

Comment (sujaypatil96) [reply assets/misc/.env.template:17]: Oh right, sorry, missed that. How about NCBI_BIOSAMPLE_DB_AS_POSTGRES_*?

Comment (turbomam) [reply assets/misc/.env.template:17]: thanks for brainstorming! how would you feel about BIOSAMPLES_RDB_PW

It's the password for a relational database of biosamples

Comment (turbomam) [reply assets/misc/.env.template:17]: I asked ChatGPT for suggestions :-)

**Review by turbomam — APPROVED** [review]

**Pull request merged**